pipeline.py

```diff
@@ -72,31 +87,76 @@ class TextProcessingPipeline:
     """
 
     def __init__(self, corpus_manager: CorpusManager):
-        pass
+        self.corpus_manager = corpus_manager
 
     def run(self):
         """
         Runs pipeline process scenario
         """
-        pass
+        for article in self.corpus_manager.get_articles().values():
+            text = article.get_raw_text()
+            tokens = self._process(text)
+            article.save_as(" ".join(map(lambda x: x.get_cleaned(), tokens)),
+                            ArtifactType.cleaned)
+            article.save_as(" ".join(map(lambda x: x.get_single_tagged(), tokens)),
+                            ArtifactType.single_tagged)
+            article.save_as(" ".join(map(lambda x: x.get_multiple_tagged(), tokens)),
+                            ArtifactType.multiple_tagged)
 
     def _process(self, raw_text: str):
         """
         Processes each token and creates MorphToken class instance
         """
-        pass
+        mystem = Mystem()
+        morph_analyzer = MorphAnalyzer()
+        # Linebreaks make Mystem slow and unresponsive.
+        tokens = []
+        for analysis in mystem.analyze(raw_text.replace("\n", " ")):
+            if "analysis" not in analysis:
+                continue
+            if not analysis["analysis"]:
+                continue
+            token = MorphologicalToken(original_word=analysis["text"])
+
+            token.normalized_form = analysis["analysis"][0]["lex"]
+            token.tags_mystem = analysis["analysis"][0]["gr"]
+            token.tags_pymorphy = morph_analyzer.parse(analysis["text"])[0].tag
+
+            tokens.append(token)
+        return tokens
 
 
 def validate_dataset(path_to_validate):
     """
     Validates folder with assets
     """
-    pass
+    path = Path(path_to_validate)
+    if not path.exists():
+        raise FileNotFoundError
+    if not path.is_dir():
+        raise NotADirectoryError
+    if not any(path.iterdir()):
+        raise EmptyDirectoryError
+    for file in path.iterdir():
+        if not file.read_bytes():
```

Comment:
file emptiness should be checked without reading it, if you want to do it, however, I do no remember any hard requirement in docs - can you point where it is stated?